src/js/controller/controller.js

```diff
@@ -133,6 +133,8 @@ Object.assign(Controller.prototype, {
             if (reason === 'clickthrough' || reason === 'interaction' || reason === 'external') {
                 _model.set('playOnViewable', false);
                 _model.off('change:playReason change:pauseReason', changeReason);
+            } else if (reason === 'playlist') {
+                _checkPauseOnViewable(_model, _model.get('viewable'));
```

Comment:
This seems risky and and also "after the fact". You should perform this logic when the reason is set. `AD_PLAY` and `AD_PAUSE` will never have a reason of 'playlist'.